### PR TITLE
chart: Add option to install VPA

### PR DESCRIPTION
**Pull request by avorima**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

I currently install a VPA for the cluster-autoscaler alongside the chart, but it would be nice to get it all from the same source.

#### Does this PR introduce a user-facing change?

```release-note
Add chart option to install VPA
```

### CODE REVIEW

File: charts/cluster-autoscaler/Chart.yaml [hunk 1]
```diff
@@ -11,4 +11,4 @@ name: cluster-autoscaler
 sources:
   - https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler
 type: application
-version: 9.25.0
+version: 9.25.1
```
Comment (gjtempleton) [inline charts/cluster-autoscaler/Chart.yaml:14]: We're adding new functionality here, so can we make this a minor version bump please?

File: charts/cluster-autoscaler/README.md [hunk 2]
```diff
@@ -309,6 +309,25 @@ Containers:
 
 Though enough for the majority of installations, the default PodSecurityPolicy _could_ be too restrictive depending on the specifics of your release. Please make sure to check that the template fits with any customizations made or disable it by setting `rbac.pspEnabled` to `false`.
 
+### VerticalPodAutoscaler
+
+The chart can install a VPA for the Deployment if needed. A VPA can help minimize wasted resources when usage spikes periodically.
+The following example snippet can be used to install a `autoscaling.k8s.io/v1` VPA.
```
Comment (gjtempleton) [inline charts/cluster-autoscaler/README.md:315]: Can we link out to the VPA docs please, there's a reasonable chance users will be reading this Readme disconnected from this repo, and may want to find out more about the VPA.

**Review by gjtempleton — COMMENTED** [review]
/assign @gjtempleton 

A couple of small requests.

File: charts/cluster-autoscaler/templates/vpa.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+{{ include "cluster-autoscaler.labels" . | indent 4 }}
+  name: {{ template "cluster-autoscaler.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+spec:
+  targetRef:
+    apiVersion: {{ template "deployment.apiVersion" . }}
+    kind: Deployment
+    name: {{ template "cluster-autoscaler.fullname" . }}
+  updatePolicy:
+    updateMode: "Auto"
```
Comment (voelzmo) [inline charts/cluster-autoscaler/templates/vpa.yaml:15]: I see people run their VPA in `updateMode: off` for a while to observe if anything weird would happen when they switched on `updateMode: auto` – so maybe this is an option you'd want to make configurable?

**Review by voelzmo — CHANGES_REQUESTED** [review]
Thanks!

Some small comments about VPA configuration settings, the API version is the only thing that I'd really recommend to change – the other two remarks are just to start a discussion. I wouldn't argue against merging this, even if you were to decide to keep these as-is.

File: charts/cluster-autoscaler/templates/vpa.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{{- if .Values.vpa.enabled -}}
+apiVersion: "autoscaling.k8s.io/{{ .Values.vpa.apiVersion }}"
```
[the remark this answers is not in the document]
Comment (avorima) [reply charts/cluster-autoscaler/templates/vpa.yaml:2]: Oh sure. I was actually trying to find out when it got released, but I must've missed those release notes.

File: charts/cluster-autoscaler/values.yaml [hunk 5]
```diff
@@ -384,3 +384,22 @@ updateStrategy: {}
...
+  # vpa.apiVersion -- APIVersion of the VerticalPodAutoscaler. The template only supports v1 and v1beta2.
+  apiVersion: v1
+  # vpa.containerPolicy -- Configure the containerPolicy of the cluster-autoscaler container. Content is rendered as YAML.
+  containerPolicy: {}
+    # minAllowed:
+    #   cpu: 50m
+    #   memory: 150Mi
+    # maxAllowed:
+    #   cpu: 100m
+    #   memory: 300Mi
```
[the remark this answers is not in the document]
Comment (avorima) [reply charts/cluster-autoscaler/values.yaml:401]: The values here were meant to illustrate the possible contents of a containerPolicy. I'll add a link to the spec instead.

I just checked some deployments and the lowest usage I saw was `2m/22Mi`. I'd say `20m/50Mi` is a good compromise that still allows some headroom. I'll add these to the example in the README and leave out the `maxAllowed`.

File: charts/cluster-autoscaler/templates/vpa.yaml (hunk 3 above)
Comment (avorima) [reply charts/cluster-autoscaler/templates/vpa.yaml:15]: I didn't think about that. I'll add it

**Review by voelzmo — APPROVED** [review]
/lgtm thanks!

**Comment by gjtempleton** [comment]
Thanks for addressing the feedback.

/lgtm
/approve

**Pull request merged**